server/events.go

```diff
@@ -289,6 +292,8 @@ func (s *Server) internalSendLoop(wg *sync.WaitGroup) {
 				c.flushClients(time.Second)
 				return
 			}
+		case <-resetChan:
+			goto RESET
```

Comment:
A close() of the this channel would then cause this function to go back to RESET, which is kind of useless since we are likely close s.quitCh too. 
Note that you can check that the channel was not actually closed:
```
case _, ok := <-resetChan:
  if ok {
      goto RESET
  }
```

Reply:
In the case you describe the go routine would start to spin. 
In either case close has to happen after s.sys.wg.Wait().

Reply:
You are right. Since we close the quitCh first and then wait on wg.Wait(), it is fine to leave the wait it is in the PR.